src/native/managed/cdacreader/tests/CodeVersionsTests.cs

```diff
@@ -63,202 +64,80 @@ public static MockMethodDesc CreateVersionable (TargetPointer selfAddress, Targe
-            return null;
-        }
-
-        private MockMethodDesc FindMethodDesc(TargetPointer targetPointer) => TryFindMethodDesc(targetPointer) ?? throw new InvalidOperationException($"MethodDesc not found for {targetPointer}");
-        private MockMethodTable FindMethodTable(TargetPointer targetPointer) => TryFindMethodTable(targetPointer) ?? throw new InvalidOperationException($"MethodTable not found for {targetPointer}");
-
-        MethodDescHandle IRuntimeTypeSystem.GetMethodDescHandle(TargetPointer targetPointer) => new MethodDescHandle(FindMethodDesc(targetPointer).Address);
-
-        bool IRuntimeTypeSystem.IsVersionable(MethodDescHandle methodDesc) => FindMethodDesc(methodDesc.Address).IsVersionable;
-        TargetCodePointer IRuntimeTypeSystem.GetNativeCode(MethodDescHandle methodDesc) => FindMethodDesc(methodDesc.Address).NativeCode;
-        TargetPointer IRuntimeTypeSystem.GetMethodDescVersioningState(MethodDescHandle methodDesc) => FindMethodDesc(methodDesc.Address).MethodDescVersioningState;
+            throw new InvalidOperationException($"No token found for 0x{token:x} in table {table}");
+        });
+    }
 
-        TargetPointer IRuntimeTypeSystem.GetMethodTable(MethodDescHandle methodDesc) => FindMethodDesc(methodDesc.Address).MethodTable?.Address ?? throw new InvalidOperationException($"MethodTable not found for {methodDesc.Address}");
-        uint IRuntimeTypeSystem.GetMethodToken(MethodDescHandle methodDesc) => FindMethodDesc(methodDesc.Address).MethodToken;
+    private void AddMethodDesc(MockMethodDesc methodDesc)
+    {
+        MethodDescHandle handle = new MethodDescHandle(methodDesc.Address);
+        mockRuntimeTypeSystem.Setup(r => r.GetMethodDescHandle(methodDesc.Address)).Returns(handle);
+        mockRuntimeTypeSystem.Setup(r => r.IsVersionable(handle)).Returns(methodDesc.IsVersionable);
+        mockRuntimeTypeSystem.Setup(r => r.GetNativeCode(handle)).Returns(methodDesc.NativeCode);
+        mockRuntimeTypeSystem.Setup(r => r.GetMethodDescVersioningState(handle)).Returns(methodDesc.MethodDescVersioningState);
+        mockRuntimeTypeSystem.Setup(r => r.GetMethodTable(handle)).Returns(() => methodDesc.MethodTable?.Address ?? throw new InvalidOperationException($"MethodTable not found for {methodDesc.Address}"));
+        mockRuntimeTypeSystem.Setup(r => r.GetMethodToken(handle)).Returns(methodDesc.MethodToken);
+    }
 
-        TypeHandle IRuntimeTypeSystem.GetTypeHandle(TargetPointer address)
+    private void AddMethodTable(MockMethodTable methodTable)
+    {
+        TypeHandle handle = new TypeHandle(methodTable.Address);
+        mockRuntimeTypeSystem.Setup(r => r.GetTypeHandle(methodTable.Address)).Returns<TargetPointer>(address =>
         {
-            ulong addressLowBits = (ulong)address & ((ulong)_target.PointerSize - 1);
-
+            // this is not quite accurate on 32 bit architectures, but it's good enough for testing
+            ulong addressLowBits = (ulong)address & 7;
```

Comment:
Should be `(ulong)address & ((ulong)_target.PointerSize - 1);`, however when we set this up the target isn't created. I guess we could create the target first and pass it in.

Reply:
You should be able to get the pointer size off of the `TargetTestHelpers` for `MockDescriptor.CodeVersions`, which we create before doing this.